### PR TITLE
fix: junit reporters crashing on failed asserts

### LINKED ISSUE #24652: Cypress v11 `cypress run` crashes when assertion in test fails
### Current behavior

If an assertion in a test fails by running `cypress run --browser chrome` the whole process fails instead of showing the result of one failing test in the console as report. 
Fails with following error:
```bash
u(...).generateDiff is not a function
TypeError: u(...).generateDiff is not a function
    at T.getTestcaseData (<embedded>:1144:21700)
    at T.<anonymous> (<embedded>:1144:19702)
    at R.emit (node:events:539:35)
    at S.emit (<embedded>:2755:16640)
    at Object.onMocha (<embedded>:2802:338633)
    at p.<anonymous> (<embedded>:2802:66203)
    at p.emit (node:events:527:28)
    at p.emitUntyped (<embedded>:2527:41392)
    at <embedded>:2527:48899
    at process.processTicksAndRejections (node:internal/process/task_queues:78:11)
```

### Desired behavior

Show the failing test result in the report table same as in v10.11

### Test code to reproduce

```js
it("Faulty test", () => {
   expect(3).to.equal(4);
}
```

### Cypress Version

11.0.0 or 11.0.1

### Node version

v16.14.2

### Operating System

macOS Ventura 13.0

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by ryanthemanuel**
- Closes #24652 

### User facing changelog

Cypress will no longer crash when using the junit reporter on failed asserts

### Additional details

It appears that we have a delicate ordering for our logic to override mocha resolution. We need to ensure that it is loaded at the appropriate time or things get loaded incorrectly. I tried moving that loading around to various places in the code base, to no avail. The only thing I could get to work was to ensure that the dependencies in this PR don't defer when they load their dependencies. It's not ideal, but we now have a system test for this scenario that should prevent things like this in the future.

### Steps to test

Write a failing test and use the `junit` reporter

### How has the user experience changed?

`cypress run` no longer crashes when an assert fails

### PR Tasks

- [X] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by mschile — APPROVED** [review]

**Review by chrisbreiding — APPROVED** [review]

File: system-tests/__snapshots__/reporters_spec.js [hunk 1]
```diff
@@ -624,11 +661,11 @@ Because this error occurred during a \`after all\` hook we are skipping the rema
...
   (Screenshots)
 
-  -  /XXX/XXX/XXX/cypress/screenshots/simple_failing_hook.cy.js/simple failing hook s     (1280x720)
+  -  /XXX/XXX/XXX/cypress/screenshots/simple_failing_hook.cy.js/simple failing hook s    (2560x1440)
      pec -- beforeEach hooks -- never gets here -- before each hook (failed).png                    
-  -  /XXX/XXX/XXX/cypress/screenshots/simple_failing_hook.cy.js/simple failing hook s     (1280x720)
+  -  /XXX/XXX/XXX/cypress/screenshots/simple_failing_hook.cy.js/simple failing hook s    (2560x1440)
      pec -- afterEach hooks -- runs this -- after each hook (failed).png                            
-  -  /XXX/XXX/XXX/cypress/screenshots/simple_failing_hook.cy.js/simple failing hook s     (1280x720)
+  -  /XXX/XXX/XXX/cypress/screenshots/simple_failing_hook.cy.js/simple failing hook s    (2560x1440)
```
Comment (BlueWinds) [inline system-tests/__snapshots__/reporters_spec.js]: Why would this change increase the snapshot size? This seems quite large for a default snapshot size.

**Review by BlueWinds — APPROVED** [review]

File: system-tests/__snapshots__/reporters_spec.js (hunk 1 above)
Comment (emilyrohrbough) [reply system-tests/__snapshots__/reporters_spec.js]: that's the size of the launched browser

File: system-tests/__snapshots__/reporters_spec.js [hunk 2]
```diff
@@ -51,6 +53,39 @@ exports['e2e reporters supports junit reporter and reporter options 1'] = `
...
+  │ Screenshots:  2                                                                                │
+  │ Video:        true                                                                             │
+  │ Duration:     X seconds                                                                        │
+  │ Spec Ran:     simple_failing.cy.js                                                             │
+  └────────────────────────────────────────────────────────────────────────────────────────────────┘
+
+
+  (Screenshots)
+
+  -  /XXX/XXX/XXX/cypress/screenshots/simple_failing.cy.js/simple failing spec -- fai    (1280x720)
```
Comment (ryanthemanuel) [inline system-tests/__snapshots__/reporters_spec.js:77]:
```suggestion
  -  /XXX/XXX/XXX/cypress/screenshots/simple_failing.cy.js/simple failing spec -- fai    (1280x720)
```

Comment (ryanthemanuel) [inline system-tests/__snapshots__/reporters_spec.js:77]:
```suggestion
  -  /XXX/XXX/XXX/cypress/screenshots/simple_failing.cy.js/simple failing spec -- fai   (1280x720)
```

File: system-tests/__snapshots__/reporters_spec.js [hunk 3]
```diff
@@ -51,6 +53,39 @@ exports['e2e reporters supports junit reporter and reporter options 1'] = `
...
+     ls1 (failed).png                                                                               
+  -  /XXX/XXX/XXX/cypress/screenshots/simple_failing.cy.js/simple failing spec -- fai    (1280x720)
```
Comment (ryanthemanuel) [inline system-tests/__snapshots__/reporters_spec.js:79]:
```suggestion
  -  /XXX/XXX/XXX/cypress/screenshots/simple_failing.cy.js/simple failing spec -- fai   (1280x720)
```

File: system-tests/__snapshots__/reporters_spec.js [hunk 4]
```diff
@@ -272,11 +309,11 @@ Because this error occurred during a \`after all\` hook we are skipping the rema
 
   (Screenshots)
 
-  -  /XXX/XXX/XXX/cypress/screenshots/simple_failing_hook.cy.js/simple failing hook s     (1280x720)
+  -  /XXX/XXX/XXX/cypress/screenshots/simple_failing_hook.cy.js/simple failing hook s    (1280x720)
```
Comment (ryanthemanuel) [inline system-tests/__snapshots__/reporters_spec.js:312]:
```suggestion
  -  /XXX/XXX/XXX/cypress/screenshots/simple_failing_hook.cy.js/simple failing hook s   (1280x720)
```

**Pull request merged**